### PR TITLE
Move scrolling to AbstractInputHandler

### LINKED ISSUE #1363: Finger touch enables eraser
**Affects versions :**
 - OS: Archlinux
 - Version of Xournal++: 59f6e91876ccca2ebeaf7ea8dd6899709fb2b333

**Describe the bug**
When using regular finger touch input to move the drawing area around the eraser tool activates. This is **working** in 1.0.12.

**To Reproduce**
1. Touch the screen with one finger

**Input Logs**
```
ALSA lib pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
connect(2) call to /dev/shm/jack-1000/default/jack_0 failed (err=Datei oder Verzeichnis nicht gefunden)
attempt to connect to server failed
connect(2) call to /dev/shm/jack-1000/default/jack_0 failed (err=Datei oder Verzeichnis nicht gefunden)
attempt to connect to server failed
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
connect(2) call to /dev/shm/jack-1000/default/jack_0 failed (err=Datei oder Verzeichnis nicht gefunden)
attempt to connect to server failed
** Message: 13:47:24.478: Event
Event type:	GDK_FOCUS_CHANGE
Source device:	GDK_SOURCE_KEYBOARD
Device Class:	INPUT_DEVICE_KEYBOARD

** Message: 13:47:24.504: Event
Event type:	GDK_ENTER_NOTIFY
Source device:	GDK_SOURCE_MOUSE
Device Class:	INPUT_DEVICE_MOUSE

** Message: 13:47:26.508: Event
Event type:	GDK_MOTION_NOTIFY
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.510: Event
Event type:	GDK_TOUCH_BEGIN
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.511: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.512: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.512: Event
Event type:	GDK_TOUCH_END
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.514: Event
Event type:	GDK_LEAVE_NOTIFY
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.551: Event
Event type:	GDK_MOTION_NOTIFY
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.555: Event
Event type:	GDK_ENTER_NOTIFY
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.555: Event
Event type:	GDK_TOUCH_BEGIN
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.556: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.557: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.558: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.559: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.571: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.583: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.650: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.651: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.652: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.652: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.653: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.654: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.666: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.681: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.690: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.701: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.714: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.725: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.737: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.749: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.763: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.773: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.784: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.798: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.809: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.821: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.833: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.846: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.857: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.868: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.881: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.892: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.904: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.918: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.930: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.940: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.952: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.965: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.975: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:26.987: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.000: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.013: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.023: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.036: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.048: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.064: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.072: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.084: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.097: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.108: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.119: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.134: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.147: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.156: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.168: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.181: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.191: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.203: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.217: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.231: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.239: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.251: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.264: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.274: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.287: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.301: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.314: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.323: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.334: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.348: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.358: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.370: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.384: Event
Event type:	GDK_TOUCH_UPDATE
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.397: Event
Event type:	GDK_TOUCH_END
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:27.402: Event
Event type:	GDK_LEAVE_NOTIFY
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:28.949: Event
Event type:	GDK_LEAVE_NOTIFY
Source device:	GDK_SOURCE_TOUCHSCREEN
Device Class:	INPUT_DEVICE_ERASER

** Message: 13:47:29.052: Event
Event type:	GDK_FOCUS_CHANGE
Source device:	GDK_SOURCE_KEYBOARD
Device Class:	INPUT_DEVICE_KEYBOARD


(xournalpp:12917): GLib-GObject-CRITICAL **: 13:47:29.054: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
** Message: 13:47:29.057: MemoryLeak: sum 0 objects.
```

**Expected behavior**
Move the drawing pane around like in previous versions.

### PR DESCRIPTION
**Pull request by lehmanju**
Fixes #1363

### CODE REVIEW

File: src/gui/inputdevices/AbstractInputHandler.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include "AbstractInputHandler.h"
 #include "InputContext.h"
 #include <gui/XournalppCursor.h>
```
Comment (Febbe) [inline src/gui/inputdevices/AbstractInputHandler.cpp]: replace <> with "" for local headers if you need to touch them.
Write a empty line between all sorts of different libraries.
Write a line between the local headers and the other xournalpp headers.

**Comment by lehmanju** [comment]
whoops, forgot to actually change that … might have tested the wrong branch

File: src/gui/inputdevices/AbstractInputHandler.cpp (hunk 1 above)
Comment (lehmanju) [reply src/gui/inputdevices/AbstractInputHandler.cpp]: no need to change those though

**Comment by lehmanju** [comment]
No longer relevant as bug description has changed significantly.

**Pull request closed without merging**